packages/itwinui-react/src/utils/functions/dev.ts

```diff
@@ -14,30 +14,27 @@ const isVitest = typeof (globalThis as any).__vitest_index__ !== 'undefined';
 
 const isUnitTest = isJest || isVitest || isMocha;
 
-let isDev = false;
-
-// wrapping in try-catch because process might be undefined
-try {
-  isDev = process.env.NODE_ENV !== 'production' && !isUnitTest;
-} catch {}
-
 /**
  * Logs message one time only in dev environments.
  *
  * @example
  * const logWarningInDev = createWarningLogger();
- * logWarningInDev("please don't use this")
+ *
+ * if (process.env.NODE_ENV === 'development') {
+ *  logWarningInDev("please don't use this")
+ * }
  */
-const createWarningLogger = !isDev
-  ? () => () => {}
-  : () => {
-      let logged = false;
-      return (message: string) => {
-        if (!logged) {
-          console.warn(message);
-          logged = true;
-        }
-      };
-    };
+const createWarningLogger =
+  process.env.NODE_ENV === 'development'
```

Comment:
Previously we were shipping this code to the user, who was responsible for handling the `NODE_ENV` env var. Whereas now, we are using SWC to handle it on our end. If you inspect the build output, you won't see `process.env.NODE_ENV` anywhere.

Good question about `isUnitTest`. I think it will use the prod bundle so should not have any warnings? (Need to confirm this)

Reply:
Did a [quick test](https://stackblitz.com/edit/vitest-dev-vitest-pd9u6c?file=src%2Fpackage.json,test%2Fbasic.test.ts) using vitest and found that it uses the `"development"` build when running unit tests.

Added back `isUnitTest` checks in [`cef8fc2`](https://github.com/iTwin/iTwinUI/pull/2131/commits/cef8fc26bf6c4c2ab3da57fbe61e0b0dd29de8df)